draft-jimenez-t2trg-mud-coap.md

```diff
@@ -129,7 +129,35 @@ TBD : IPv4, coap://, ...
 
 ## Neighbor Discovery Protocol (NDP)
 
-IPv6 hosts do not require DHCP to get access to the default gateway. {{!RFC4861}} can also be used to advertise the MUD URL.
+IPv6 hosts do not require DHCP to get access to the default gateway.
+Using NDP {{!RFC4861}} and Stateless Address Autoconfiguration (SLAAC) {{!RFC4862}}, nodes can configure global addresses on their own based on prefixes contained in NDP Router Advertisements (RAs).
+Therefore, DHCPv6 is typically not necessary for IPv6 hosts, which is a problem for the distribution of MUD URLs for these kinds of devices.
+
+To solve this issue, this document introduces a new option for the NDP which makes it possible to emit a MUD URL in Router Solicitations and Neighbor Advertisements.
+This option has the following format:
+
+~~~~
+0                   1                   2                   3
+0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|         Type                  |              Length           |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                            MUD URL                            |
+|                                                               |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
```

Comment:
I just noticed that the option format is a bit off. Both type and length should have a length of 8-bit (cf. for example the CoRE RD [Resource Directory Address Option (RDAO)](https://datatracker.ietf.org/doc/html/draft-ietf-core-resource-directory#section-4.1.1) for NDP. I will provide a fix for that soon.

Reply:
Fixed :)